test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env tarantool
+
+local fiber = require('fiber')
+local tap = require('tap')
+local popen = require('popen')
+
+--
+-- gh-2717: tarantool console quit on sigint
+--
+
+local TARANTOOL_PATH = arg[-1]
+
+local res = tap.test('gh-2717', function(test)
+    print(TARANTOOL_PATH)
+
+    test:plan(2)
+    local prompt = 'tarantool> '
+    local ph = popen.shell(TARANTOOL_PATH .. ' -i', 'r')
+    assert(ph.pid, 'popen error while executing ' .. TARANTOOL_PATH)
+
+    fiber.sleep(0.5)
+    ph:signal(popen.signal.SIGINT)
+    fiber.sleep(0.5)
+
+    local output = ph:read()
+    test:like(ph:info().status.state, popen.state.ALIVE, 'process is still alive after SIGINT ')
+
+    ph:close()
+    print(output)
+    test:like(output, prompt .. '^C\n' .. prompt, 'SIGINT doesn\'t kill tarantool in interactive mode ')
+    ph:close()
+end)
+os.exit(res and 0 or 1)
```

Comment:
`res` is a test object, it is never `nil` or `false`. See how other tests call `os.exit()`.

Reply:
It seems, I was looking into incorrect example of test at the beginning. Now it is fixed.